test/fixtures/rules/jsx-a11y/jsx-a11y.js

```diff
@@ -0,0 +1,360 @@
+(() => {
+    // Fail
+    // Disabling the following rules to showcase error
+    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
+    const bad = () => <div onClick={ console.log() } />;
+
+    // Succeed
+    const good = () => <div onClick={ console.log() } onKeyDown={ console.log() } role="link" tabIndex="0" />;
+})();
+
+// '/heading-has-content': Enforce heading (h1, h2, etc) elements contain accessible content.
+(() => {
+    // Fail
+    const bad1 = () => <h1 />;
+    const bad2 = () => <h1><p aria-hidden> Hidden </p></h1>;
+
+    // Succeed
+    const good = () => <h1><p> Visible </p></h1>;
+})();
+
+// '/html-has-lang': Enforce <html> element has lang prop.
+(() => {
+    // Fail
+    const bad1 = () => <html />;
+
+    // Succeed
+    const good = () => <html lang="pt" />;
+})();
+
+// '/iframe-has-title': Enforce iframe elements have a title attribute.
+(() => {
+    // Fail
+    const bad = () => <iframe />;
+
+    // Succeed
+    const good = () => <iframe title="has a title" />;
+})();
+
+// '/img-redundant-alt': Enforce <img> alt prop does not contain the word "image", "picture", or "photo".
+(() => {
+    // Fail
+    const bad = () => <img alt="this image is a picture of a photo" />;
+
+    // Succeed
+    const good = () => <img alt="Example of a negative after developing a roll of film." />;
+})();
+
+// '/interactive-supports-focus': Enforce that elements with interactive handlers like onClick must be focusable.
+(() => {
+    // Fail
+    const bad = () => <span onClick={ console.log() } onKeyDown={ console.log() } role="button">Link</span>;
+
+    // Succeed
+    // eslint-disable-next-line jsx-a11y/anchor-has-content
+    const good = () => <span onClick={ console.log() } onKeyDown={ console.log() } role="button" tabIndex="0">Link</span>;
+})();
+
+// '/label-has-associated-control': Enforce that a label tag has a text label and an associated control.
+(() => {
+    // Fail
+    const bad1 = () =>
+        (<div>
+            <label>
+                Link
+            </label>
+            <input />
+        </div>);
+    const bad2 = () =>
+        (<div>
+            <label>
+                Link
+            </label>
+        </div>);
+
+    // Succeed
+    const good1 = () =>
+        (<div>
+            <label>
+                Link
+                <input />
+            </label>
+        </div>);
+    const good2 = () =>
+        (<div>
+            <label htmlFor="input">
+                Link
+            </label>
+            <input id="input" />
+        </div>);
+})();
+
+// '/lang': Enforce lang attribute has a valid value.
+(() => {
+    // Fail
+    const bad = () => <html lang="PT" />;
+
+    // Succeed
+    const good = () => <html lang="pt" />;
+})();
+
+// '/media-has-caption': Enforces that <audio> and <video> elements must have a <track> for captions.
+(() => {
+    // Fail
+    const bad1 = () => <audio />;
```

Comment:
This is not failing because we disabled this rule.

Reply:
Hmm, you're right, do you think i should document this in some other way? 

My reasoning for keeping it this way is, this file should reflect our options at any given point in time, and having the written example of a bad practice lacking the warning is, in my view, a proper reflection of our choices.

Also, though a bit more pedantically, i think we can say that we're testing for is the presence of the warnings, so, in effect, the fact that this is not failing means that the test is passing as it was expected to.

Reply:
If it's not enforced you should not document it :)

Reply:
@satazor @acostalima any opinions on this?

Reply:
My question for the time being would be why are we disabling this rule? 🤔

Reply:
What are we going to do about this? 🤔

Reply:
I'm fine with enabling it but as a warning.

Reply:
Resolving as this rule was activated as a warning.